### DISCUSSION TITLE
Businesses acting as truck stations.

### DISCUSSION THREAD

**Discussion by ohlidalp** [voxeltycoon/issues#960] [General] [answered]
Hello.

As my empire expands, I feel I have less and less time to do actual surveying and strategizing - Instead most of my time is eaten by these 2 mechanical tasks:
* **Supply new business** - in 90% cases (Diesel engine 1/2) this involves creating a truck route with 1-2 vehicles. The only strategic choice is selecting/building a source. Actions to perform: 1. open bottom Roads menu, 2. open Stations submenu, 3. pick station, 4. place station, 5. fix road, 6. scroll to destination, 7. pick station again, 8. place station again, 9. fix road again, 10. locate depot, 11. click depot, 12. click "buy", 13. pick vehicle, 14. assign cargo type (so that route-name generator works later) 15. click Buy, 16. bring up orders menu, 17. click AddOrder, 18. click the starting station + select order, 19. click AddOrder again (you could hold shift but usually you must also scroll map to locate the station again), 20. click the destinations station. 21. click ReleaseAll (because it's bigger than individual release switch, easier to find with mouse), 22. click SaveRoute - textbox appears, 23. click OK on the textbox,  24. hit DELETE to clean up the UI mess.
* **Improve supply to existing business** - in 90% cases this means adding 1-2 vehicles to existing route. Also most of the time the truck stop by the business is hard to find with naked eye due to city around it. Also quite often there are manufacturing issues to resolve. Actions: 1. click the business to bring up the BusinessWindow 2. click Routes tab. 3. Click the truck stop icon - you're zoomed there 3B: click the truck stop to bring up window. 4. click one truck icon to bring it's orders menu. 5. click the source station to check manufacturing problems, 6. assuming manufacturing is OK, locate a depot to add more vehicles. 7. click the depot, 8. click the Copy tool 9. click the truck window you have opened. 10. click ReleaseAll. 11. Hit DELETE to clean up the UI mess.

In Transport Tycoon, my primary monitoring device was station list widow, which showed big colored bar for resources in stock and thin red/green transport rating bar. Worse rating meant lower production and wasted potential. In Voxel Tycoon, I don't have any such window, I must survey the map by scrolling and look for the colored dots in demand indicators which show Satisfaction rank.

I spent a while thinking how to help myself with modding (a business list window? A SupplyRouteManager which remembers which saved route supplies what business?). Now that I have a consistent idea, I realize it could be described here as a suggestion:

_**Since businesses always spawn by a road, why not make them acting truck stops.**_

 This would make supplying new business much quicker - the player would just order the truck to the business. Unload order would be redundant since businesses are only sinks. Also, this would form a direct link between the business and the route which supplies it, which would make it possible to show the Satisfaction rating in the fleet window. This would make monitoring businesses a completely different experience - player could just use the CopyTruck tool on the fleet window itself where needed.

**Comment by andrewpey** [comment, marked as answer]
Thanks for the detailed argumentation! Actually, we have plans for that.
